src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -218,7 +219,27 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                             targetsList = await GetTargetsListAsync(assetsFilePath, token);
                         }
 
-                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, token) : new Dictionary<string, TransitiveEntry>();
+                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
+                        List<PackageReference> installedProjectReferences = new List<PackageReference>(projectReferences.Count());
+
+                        // Since ProjectReferences are not in the dependencies section of the package spec (installed packages), we need to manually add them
+                        // to the installed packages list to compute transitive origins
+                        foreach (var projectReference in projectReferences)
+                        {
+                            var packageReference = new PackageReference(
```

Comment:
We probably should have a new type for modeling project references. Please see my comment in the one-pager for details: https://github.com/NuGet/Client.Engineering/pull/2934#discussion_r1680165077

Reply:
@nkolev92 Can correct me here, but my understanding is that `<ProjectReferences>` [has a duality](https://github.com/NuGet/Home/wiki/PackageReference-Specification) with `<PackageReferences>` that's why I consider it as an option.

Reply:
From restore's point of view, they're similar (I guess the goal was to treat them the same, but there are some significant gaps). But basically, transitive packages (and some things like NoWarn) flow through package dependencies just the same as ProjectReferences.

I haven't looked at the code changes in this PR, so I'm just basing my comments on this thread's comments, but consider what does it mean if a project reference appears in the Package Manager UI. What will the version list contain. What if there's a package on one of the feeds with the same ID, should PM UI allow you to "install" the package (change ProjectReference to a PackageReference). Can you uninstall a ProjectReference? If so, do we need to list all solution projects somewhere so you can add project references via PM UI?

The top of the PackageReference spec says that the context is about migrating from project.json and xproj to MSBuild files. I think taking a statement about project & package duality from this document and extrapolating it to "show projects in Package Manager UI's packages list" is taking that statement out of context.

Reply:
`ProjectReferences` are not going to be displayed in the PM UI. 

The current implementation to gather transitive packages and their origin is to look at installed packages from the `PackageSpec` and add their dependencies. The problem here is these installed packages don't have the information for project references. That's why I'm adding them manually to a copy of the installed packages list just to gather their dependencies and calculate the origins.